### DISCUSSION TITLE
Re-creating tasmota32-bluetooth.bin with DS18x20

### DISCUSSION THREAD

**Discussion by dgoadby** [arendst/Tasmota#17951] [Q&A] [answered]
I am using tasmota32-bluetooth.bin (core 2.0.5.3, model: 01 ESP32-DevKit) and I want to add DS18x20 sensors to the build. 

Now it might not be possible due to the final binary size but - nothing ventured nothing gained!

What would be useful is a copy of the "my_user_config.h" that was used to build it. Then I can use "user_config_override.h" to add the DS18x20 override. 

In fact a copy of the configuration files for all of the build would be useful. They must be located somewhere as the binaries already exist.

I have already setup the PlatformIO compiler environment and built a standard binary. With the "my_user_config.h" used for Bluetooth and my own "user_config_override.h" I should be able to compile a Bluetooth system with DS18x20 included. 

If it works then I will happily offer my binary to the downloadable list.

**Comment by barbudor** [comment, marked as answer]
The coniguration for each build variant starts from `platformio_tasmota_env.ini` and `_env32.ini` (the latter in your case)
Each build environnement is defined for
- compilation options
- libraries and linking options

Part of the compile options is a `-DFIRMWARE_XXX` option which define the macro `FIRMWARE_XXX`

Then the list of #defines for each are in `tasmota/include/tasmota_configurations.h` or `_ESP32.h`

You will have to create your own environment to get the libraries and link options
And then copy what you need into user_config_override.h.
All this is documeted in Compile Your Build

**Reply by sfromis** [reply]
Note that the precompiled `tasmota32-bluetooth.bin` **already** includes support for DS18x20.

**Reply by dgoadby** [reply]
Thanks sfromis that is interesting and, if true, will save me from a steep leaning curve! 

However, the wiki entry for the DS18B20 does say: 

![image](https://user-images.githubusercontent.com/1812006/219014721-137c2501-b4c6-47f1-a7e4-a09d518b17dd.png)

Assuming that you are correct then I now have to figure out why it does not work on my target system. I only tried one sensor but I did try many different I/O pins. 

I will now try another sensor and also add a 4k7 resistor so that I am not reliant on the weak internal pull-up. I also think it is time to get the scope out and see if the I/O pin has any activity on it. 

Thanks to barbudor for helping with the build process.  I have been delving deeper into the code and it is a very impressive suite which we should be proud of. And, as a recent and grateful user, I have sent a donation too.

**Reply by sfromis** [reply]
That clause does not say **only** :wink:  Anyway, such minute details are not consistently changed in all of docs. People can submit proposals for change.

I actually verified that the gpio dropdown on a device flashed with `tasmota32-bluetooth.bin` included `DS18x20`. And yes, my experience with DS18B20 on ESP32 is that I need to have external pullup.

**Reply by dgoadby** [reply]
Thanks for verifying. 

I routinely add the 4k7 and I should have disregarded the comment that one device would be ok. 

Thanks again.
